src/messageComposer/messageComposer.ts

```diff
@@ -449,58 +448,73 @@ export class MessageComposer {
         !draft ||
         !!draft.parent_id !== !!this.threadId ||
         draft.channel_cid !== this.channel.cid
-      )
+      ) {
         return;
+      }
 
       this.logDraftUpdateTimestamp();
+
       if (this.compositionIsEmpty) {
         return;
       }
+
       this.clear();
     }).unsubscribe;
 
   private subscribeTextComposerStateChanged = () =>
-    this.textComposer.state.subscribe((nextValue, previousValue) => {
-      if (previousValue && nextValue.text !== previousValue?.text) {
+    this.textComposer.state.subscribeWithSelector(
+      ({ text }) => [text] as const,
+      ([currentText], previousSelection) => {
+        // do not handle on initial subscription
+        if (typeof previousSelection === 'undefined') return;
+
         this.logStateUpdateTimestamp();
+
         if (this.compositionIsEmpty) {
           this.deleteDraft();
           return;
         }
-      }
-      if (!this.linkPreviewsManager.enabled || nextValue.text === previousValue?.text)
```

Comment:
Removed this part:

`nextValue.text === previousValue?.text`

as selected text is already being compared against previous value internally and unless it differs between state changes, the handler won't run.